compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -387,6 +387,12 @@ private BuildOptions parseBuildOptions() {
                 String version = getStringValueFromDependencyNode(dependencyNode, VERSION);
                 String repository = getStringValueFromDependencyNode(dependencyNode, REPOSITORY);
 
+                if (repository != null && !repository.equals(ProjectConstants.LOCAL_REPOSITORY_NAME)) {
+                    reportDiagnostic(dependencyNode,
+                            "invalid value '" + repository + "'",
```

Comment:
Shall we check if this validation can be added to the ballerina-toml-schema? IFAIK, we can give an allowed pattern

Reply:
Ref: https://github.com/ballerina-platform/ballerina-lang/blob/a91fcae07236692df6c0b045e0fde95525f54b95/compiler/ballerina-lang/src/main/resources/ballerina-toml-schema.json#L115

Reply:
will add this

Reply:
added this and removed the check from Manifest builder